### ISSUE TITLE
Mouse Wheel Lag

### ISSUE THREAD

**Issue by biztyke** [folivoraAI/BetterTouchTool#298]
**#### Description of your bug/question/feature request...:**

In Ableton Live, I’ve mapped the + (zoom in) and - (zoom out) keyboard keys to my "wheel up" and "wheel down" actions so that I can (finally!) use the mouse wheel to zoom in and out—a feature that is native in practically all other audio, photography, video and text apps yet inexplicably doesn't exist in Ableton. While the above mapping setup works, there is quite a bit of lag and unresponsive when doing so.

Meanwhile, using another mapping app called USB Overdrive, the zooming is extremely smooth and rapid. Likewise, on my Windows 7 machine, I have gone through the same process with a open-source program called autohotkey, and everything works perfectly there too (using the same mouse).

Therefore, am I doing something wrong or is this an issue with BetterTouchTool not refreshing quickly or smoothly enough? I attached a ZIP file showing the difference between zooming in and out using BetterTouchTool and USB Overdrive.

--------------

**What kind of Mac are you using?**

MacBook Pro Retina, 13-inch, Late 2012 / 2.5 GHz Intel Core i5 / 8 GB 1600 MHz DDR3 / Intel HD Graphics 4000 1024 MB

--------------

**Which input device (e.g. MacBook Trackad, Magic Mouse, Magic Trackpad Touch Bar) is affected by this?**

Normal mouse

--------------

**Which versions of macOS and BetterTouchTool are you running?**

OS X 10.9.5 (13F1911) / BetterTouchTool V 1.93

--------------

**Other information:** (e.g. stacktraces, related issues, screenshots,suggestions how to fix, etc)

[Wheel Zoom Comparison.zip](https://github.com/fifafu/BetterTouchTool/files/721648/Wheel.Zoom.Comparison.zip)

**Comment by biztyke** [comment]
Hello?

**Comment by fifafu** [comment]
Hey,

sorry BTT's scroll wheel implementation is currently not suited for something like this. It's intended to be used with discrete action. Therefore to prevent triggering the same action multiple times by accidentally scrolling to far it pauses recognition for a bit after triggering an action.

I'll check if I can add a setting to make this configurable!

Best,
Andreas

**Comment by biztyke** [comment]
Hi Andreas,

Good to hear from you! An option to control this would be amazing, as it would open up the mouse wheel to all sorts of possibilities. I’m sure there are programs other than Ableton that currently lack zoom options, so BTT would instantly become a saviour (haha) in all those situations.

Even though I’m still using OS X 10.9.5, I’d definitely update my OS if you add this feature in a future update. It’s that important to me, as it speeds up my workflow tremendously. Since I already love and fully support your product, I would really rather not pay to get a full license for another program like USB Overdrive or Steer Mouse just to get this one feature. BTT is a superior app in every other way! 

Thanks in advance,
Johnny

**Comment by hockey32** [comment]
Has this been developed yet?

**Comment by biztyke** [comment]
Not as far as I know... I really do hope this feature does get some love in the near future. It seems like an easy fix, yet it would be tremendously useful to those who need it.

**Comment by abertheham** [comment]
Just to make sure the lag you're experiencing is Abelton-related and not a result of BTT, have you checked this setting?  I use my scrollwheel to move between tabs and in order to make it switch with each click I had to drop the "Tilt-wheel repeat delay."

I know Abelton is much more demanding than switching tabs, but figured it was worth asking...  The setting is relatively well buried in Advanced settings.

Hope it helps!
![image](https://cloud.githubusercontent.com/assets/18314254/26784297/7fe3106a-49c2-11e7-8821-010bd419f7b9.png)

**Comment by biztyke** [comment]
Yes, I have tried playing with this shortly after I first installed BTT, but unfortunately the issue still persists. It can’t be an issue with Ableton because two other programs that are similar to BTT both zoom perfectly using the scroll wheel. Unfortunately, neither one of those programs allows modifiers to be used in conjunction with the mouse wheel. And without a modifier I lose the ability to scroll up and down through tracks, which is obviously a critical function. Thanks for the suggestion though—hopefully Fifafu sees this thread and incorporates this into a future update!

Cheers,

Johnny
English Tutor / Editor / Writer

On Jun 5, 2017, at 5:43 AM, abertheham <notifications@github.com<mailto:notifications@github.com>> wrote:


Just to make sure the lag you're experiencing is Abelton-related and not a result of BTT, have you checked this setting? I use my scrollwheel to move between tabs and in order to make it switch with each click I had to drop the "Tilt-wheel repeat delay."

I know Abelton is much more demanding than switching tabs, but figured it was worth asking... The setting is relatively well buried in Advanced settings.

Hope it helps!
[image]<https://cloud.githubusercontent.com/assets/18314254/26784297/7fe3106a-49c2-11e7-8821-010bd419f7b9.png>

**Comment by biztyke** [comment]
Yes, I have tried playing with this shortly after I first installed BTT, but unfortunately the issue still persists. It can’t be an issue with Ableton because two other programs that are similar to BTT both zoom perfectly using the scroll wheel. Unfortunately, neither one of those programs allows modifiers to be used in conjunction with the mouse wheel. And without a modifier I lose the ability to scroll up and down through tracks, which is obviously a critical function. Thanks for the suggestion though—hopefully Fifafu sees this thread and incorporates this into a future update!
